src/daft-core/src/array/ops/date.rs

```diff
@@ -108,6 +108,78 @@ impl TimestampArray {
         ))
     }
 
+    pub fn time(&self, timeunit_for_cast: &TimeUnit) -> DaftResult<TimeArray> {
+        let physical = self.physical.as_arrow();
+        let DataType::Timestamp(timeunit, tz) = self.data_type() else {
+            unreachable!("Timestamp array must have Timestamp datatype")
+        };
+        let tu = timeunit.to_arrow();
+        let timeunit_for_cast = if timeunit_for_cast == &TimeUnit::Nanoseconds {
+            TimeUnit::Nanoseconds
+        } else {
+            TimeUnit::Microseconds // default to microseconds
+        };
```

Comment:
We should return an error if any time unit is received other than microseconds or nanoseconds, rather than transparently falling back to microseconds.
```suggestion
        if !matches!(timeunit_for_cast, TimeUnit::Microseconds | TimeUnit::Microseconds) {
            return Err(DaftError::ValueError(format!("Can't cast a Timestamp array to a Time array with time unit {timeunit_for_cast}, only Microseconds and Nanoseconds are supported."));
        }
```